docs/topics/azure-api-throttling.md

```diff
@@ -0,0 +1,550 @@
+See "man sudo_root" for details.
+
+azureuser@k8s-master-31453872-2:~$ sudo systemctl stop kubelet
+azureuser@k8s-master-31453872-2:~$ sudo systemctl status kubelet
+● kubelet.service - Kubelet
+   Loaded: loaded (/etc/systemd/system/kubelet.service; enabled; vendor preset: enabled)
+   Active: inactive (dead) since Wed 2020-06-24 17:56:30 UTC; 5s ago
+ Main PID: 5243 (code=exited, status=0/SUCCESS)
+
+Jun 24 17:19:50 k8s-master-31453872-2 kubelet[5243]: I0624 17:19:50.498593    5243 container_manager_linux.go:457] [ContainerManager]: Discovered runtime cgroups name: /system.slice/docker.service
+Jun 24 17:24:50 k8s-master-31453872-2 kubelet[5243]: I0624 17:24:50.499897    5243 container_manager_linux.go:457] [ContainerManager]: Discovered runtime cgroups name: /system.slice/docker.service
+Jun 24 17:29:50 k8s-master-31453872-2 kubelet[5243]: I0624 17:29:50.500582    5243 container_manager_linux.go:457] [ContainerManager]: Discovered runtime cgroups name: /system.slice/docker.service
+Jun 24 17:34:50 k8s-master-31453872-2 kubelet[5243]: I0624 17:34:50.501119    5243 container_manager_linux.go:457] [ContainerManager]: Discovered runtime cgroups name: /system.slice/docker.service
+Jun 24 17:39:50 k8s-master-31453872-2 kubelet[5243]: I0624 17:39:50.501761    5243 container_manager_linux.go:457] [ContainerManager]: Discovered runtime cgroups name: /system.slice/docker.service
+Jun 24 17:44:50 k8s-master-31453872-2 kubelet[5243]: I0624 17:44:50.505077    5243 container_manager_linux.go:457] [ContainerManager]: Discovered runtime cgroups name: /system.slice/docker.service
+Jun 24 17:49:50 k8s-master-31453872-2 kubelet[5243]: I0624 17:49:50.505539    5243 container_manager_linux.go:457] [ContainerManager]: Discovered runtime cgroups name: /system.slice/docker.service
+Jun 24 17:54:50 k8s-master-31453872-2 kubelet[5243]: I0624 17:54:50.505996    5243 container_manager_linux.go:457] [ContainerManager]: Discovered runtime cgroups name: /system.slice/docker.service
+Jun 24 17:56:30 k8s-master-31453872-2 systemd[1]: Stopping Kubelet...
+Jun 24 17:56:30 k8s-master-31453872-2 systemd[1]: Stopped Kubelet.
+azureuser@k8s-master-31453872-2:~$ exit
+logout
+Connection to k8s-master-31453872-2 closed.
+```
+
+## Update controller-manager runtime
+
+Depending on the time window (5 minute or one hour) of your API throttling violation, you may have to wait at least 30 minutes to reliably restart your control plane runtime without being throttled once again. If you're unsure about the nature of your throttling violations (it can be opaque to debug, but log entries in the active controller-manager pod can show you some practical throttling evidence; you'll have to do that prior to shutting down the kubelet systemd job as demonstrated above) then waiting the full 30 minutes is the most likely, though conservative, tactic to take, to prevent yourself from thrashing around this process many times, re-engaging API throttling behaviors *while* you're remediating.

```

Comment:
```suggestion
Depending on the time window (5 minute or one hour) of your API throttling violation, you may have to wait at least 30 minutes to reliably restart your control plane runtime without being throttled once again. If you're unsure about the nature of your throttling violations, then waiting the full 30 minutes is the most conservative tactic to take.

(it can be opaque to debug the nature of the throttling violation, but log entries in the active controller-manager pod can show you some practical throttling evidence. You'll have to do that prior to shutting down the kubelet systemd job as demonstrated above.)
```